pkg/api/v1/resource/helpers.go

```diff
@@ -135,14 +160,39 @@ func PodLimits(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 		}
 		addResourceList(limits, container.Resources.Limits)
 	}
+
+	var sidecarResources v1.ResourceList
+	if opts.SidecarContainersEnabled {
+		sidecarResources = v1.ResourceList{}
+	}
+	initContainerLimits := v1.ResourceList{}
 	// init containers define the minimum of any resource
 	for _, container := range pod.Spec.InitContainers {
+		containerLimits := container.Resources.Limits
+		// Is the sidecar feature gate enabled and the init container marked as a sidecar?
+		if opts.SidecarContainersEnabled {
+			if container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
+				addResourceList(limits, containerLimits)
+
+				// track our cumulative sidecar resources
+				addResourceList(sidecarResources, containerLimits)
+				containerLimits = sidecarResources
+			} else {
+				tmp := v1.ResourceList{}
+				addResourceList(tmp, containerLimits)
+				addResourceList(tmp, sidecarResources)
+				containerLimits = tmp
+			}
```

Comment:
```suggestion
	for _, container := range pod.Spec.InitContainers {
		containerLimits := container.Resources.Limits.DeepCopy()
		// Is the sidecar feature gate enabled and the init container marked as a sidecar?
		if opts.SidecarContainersEnabled {
			if container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
				addResourceList(limits, containerLimits)
			}
			// track our cumulative sidecar resources
			addResourceList(containerLimits, sidecarResources)
```

Maybe this looks more clear. Correct me if I am wrong.
(similar to the other 3 functions.)

Reply:
I tried that, but I think it makes it worse.  You end up needing to do a deep-copy of the map in a few other places.